core/extension/gdextension.cpp

```diff
@@ -520,6 +544,39 @@ Ref<Resource> GDExtensionResourceLoader::load(const String &p_path, const String
 
 	String entry_symbol = config->get_value("configuration", "entry_symbol");
 
+	uint32_t compatibility_minimum[3] = { 0, 0, 0 };
+	if (config->has_section_key("configuration", "compatibility_minimum")) {
+		String compat_string = config->get_value("configuration", "compatibility_minimum");
+		Vector<int> parts = compat_string.split_ints(".");
+		for (int i = 0; i < parts.size(); i++) {
+			if (i >= 3) {
+				break;
+			}
+			if (parts[i] >= 0) {
+				compatibility_minimum[i] = parts[i];
+			}
+		}
+	}
+	if (compatibility_minimum[0] < 4) {
+		compatibility_minimum[0] = 4;
+	}
+
+	bool compatible = true;
+	if (VERSION_MAJOR < compatibility_minimum[0]) {
+		compatible = false;
+	} else if (VERSION_MINOR < compatibility_minimum[1]) {
+		compatible = false;
+	} else if (VERSION_PATCH < compatibility_minimum[2]) {
+		compatible = false;
+	}
+	if (!compatible) {
+		if (r_error) {
+			*r_error = ERR_INVALID_DATA;
+		}
+		ERR_PRINT("GDExtension is incompatible with this version of Godot: " + p_path);
```

Comment:
It would be good to print the minimum compatible version here too, e.g. for users who just download a GDExtension and don't know where that information can otherwise be found.

Reply:
Great idea! I'll get in my next push